lib/opentelemetry/trace/simple_span_processor.lua

```diff
@@ -0,0 +1,40 @@
+--------------------------------------------------------------------------------
+-- The simple span processor immediately exports spans as they finish. It does
+-- not batch spans or process the spans in a background thread (as the batch
+-- span processor does via ngx.timer.at). It is intended to be used for
+-- debugging.
+--------------------------------------------------------------------------------
+
+local _M = {
+}
+
+local mt = {
+    __index = _M
+}
+
+--------------------------------------------------------------------------------
+-- create a simple span processor.
+--
+-- @param exporter      an exporter that will be used to send span data to its
+--                      destination.
+-- @return              processor
+--------------------------------------------------------------------------------
+function _M.new(exporter)
+    return setmetatable({
+        exporter = exporter,
+    }, mt)
+end
+
+function _M.on_end(self, span)
+    if not span.ctx:is_sampled() or self.closed then
```

Comment:
Should we set `self.closed = true` in `shutdown`?

Reply:
Yeah, my mistake.

Reply:
0e80104cc40de140a3b26f439e3a3f7591f20af4